### PR TITLE
feat: navigation

### PR DESCRIPTION
**Pull request by tracy-french**
# Description

This PR enables users to navigate the application.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Build ran successfully locally
- [x] e2e test written and passed

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Legal

This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: apps/client/src/router/routes.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+export const ROUTES: RouteObject[] = [
+  {
+    path: '/',
+    element: <Root />,
+    children: [
+      {
+        path: '/login',
+        element: <LoginPage />,
+        handle: createRouteMatchHandle(LOGIN_PAGE_TYPE),
```
Comment (NorbertNader) [inline apps/client/src/router/routes.tsx:26]: This is pretty cool. I am a little worried we are introducing a non standard property to the route which is not part of the `RouteObject`. Is there any other way we could map the pages to certain page types? As I understand this is to support  the contentType in AppLayout?

File: apps/client/src/dashboard-list/dashboard-list.page.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { screen } from '@testing-library/react';
+
+import { renderWithRouter } from '../test/render-with-router';
+
+describe('<DashboardListPage />', () => {
+  test('as a user, I want to view a list of my dashboards', () => {
```
Comment (NorbertNader) [inline apps/client/src/dashboard-list/dashboard-list.page.spec.tsx:6]: I love this test naming convention. I feel we should make this a standard practice.

File: tests/navigation.spec.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { test, expect } from '@playwright/test';
+
+test.describe('Navigation', () => {
```
Comment (NorbertNader) [inline tests/navigation.spec.ts:3]: nice!

**Review by NorbertNader — APPROVED** [review]

File: apps/client/src/router/routes.tsx (hunk 1 above)
Comment (NorbertNader) [inline apps/client/src/router/routes.tsx:26]: cool use of handle

File: apps/client/src/dashboard-list/dashboard-list.page.spec.tsx (hunk 2 above)
Comment (tracy-french) [reply apps/client/src/dashboard-list/dashboard-list.page.spec.tsx:6]: Thanks man! When I write test cases like user stories, I feel much more confident that I tested the right things. Would love to consider making it the standard. :)

File: apps/client/src/dashboard/dashboard.page.tsx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import { useMatches } from 'react-router-dom';
+
+export const DashboardPage = () => {
+  const matches = useMatches();
+  const dashboardId = matches[matches.length - 1]?.params.id;
```
Comment (chejimmy) [inline apps/client/src/dashboard/dashboard.page.tsx:5]: it the last match always going to the one holding the params?

File: apps/client/src/root/root.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+  return (
+    <Page
+      content={<Outlet />}
+      pageType={pageType}
+      location={location}
+      setLocation={navigate}
+    />
+  );
+};
```
Comment (chejimmy) [inline apps/client/src/root/root.tsx:23]: this is clean 👏

**Review by chejimmy — APPROVED** [review]
LGTM! TY 🙏 for backfilling the tests as well

File: apps/client/src/root/root.tsx (hunk 5 above)
Comment (tracy-french) [reply apps/client/src/root/root.tsx:23]: Trying my best to make the lower-level components stateless and without dependencies that will make them difficult to test!

File: apps/client/src/dashboard/dashboard.page.tsx (hunk 4 above)
Comment (tracy-french) [reply apps/client/src/dashboard/dashboard.page.tsx:5]: I plan to remove the dependency on useMatches on the page in a follow-up, but from my observation, all matches hold the params.

**Pull request merged**